Android/app/src/main/java/app/intra/socks/OverrideSocksHandler.java

```diff
@@ -0,0 +1,366 @@
+        reply = ServerReply.TTL_EXPIRED;
+      } else {
+        reply = ServerReply.GENERAL_SOCKS_SERVER_FAILURE;
+      }
+      logger.info("SESSION[{}] connect {} [{}] exception:{}", session.getId(), new
+          InetSocketAddress(remoteServerAddress, remoteServerPort), reply, e.getMessage());
+    }
+
+    CommandResponseMessage responseMessage =
+        new CommandResponseMessage(VERSION, reply, bindAddress, bindPort);
+    session.write(responseMessage);
+    if (reply != ServerReply.SUCCEEDED) {
+      session.close();
+      return;
+    }
+
+    // Code prior to this point is essentially unmodified from sockslib.  At this point, sockslib
+    // normally creates a SocketPipe to pass the data.  We change the behavior here to allow retry.
+
+    final OutputStream sessionDownload = session.getOutputStream();
+    final InputStream sessionUpload = session.getInputStream();
+    InputStream socketDownload = socket.getInputStream();
+    OutputStream socketUpload = socket.getOutputStream();
+
+    final StreamPipe upload = new StreamPipe(sessionUpload, socketUpload, UPLOAD);
+    upload.setBufferSize(BUFFER_SIZE);
+    StreamPipe download = new StreamPipe(socketDownload, sessionDownload, DOWNLOAD);
+    download.setBufferSize(BUFFER_SIZE);
+
+    StatsListener listener = null;
+    try {
+      listener = runPipes(upload, download, remoteServerPort, SIMULATE_RESET && remoteServerPort == HTTPS_PORT);
+      if (remoteServerPort == HTTPS_PORT && listener.downloadBytes == 0 && listener.uploadBytes > 0
+          && listener.error != null && RESET_MESSAGE.equals(listener.error.getMessage())
+          && listener.uploadBuffer != null) {
+        socket.close();
+        download.stop();
+        socket = new Socket(remoteServerAddress, remoteServerPort);
+        socket.setTcpNoDelay(true);
+        socketDownload = socket.getInputStream();
+        socketUpload = socket.getOutputStream();
+
+        download = new StreamPipe(socketDownload, sessionDownload, DOWNLOAD);
+        download.setBufferSize(BUFFER_SIZE);
+
+        // upload is blocked in a read from sessionUpload, which cannot be canceled.
+        // However, we have modified StreamPipe to have a setter for the destination stream, so
+        // that the next transfer will be uploaded into the new socket instead of the old one.
+        upload.setDestination(socketUpload);
+
+        listener = splitRetry(upload, download, socketUpload, listener);
+      }
+    } catch (InterruptedException e) {
+      // An interrupt is a normal lifecycle event and indicates that we should perform a clean
+      // shutdown.
+    }
+
+    // Terminate and release both sockets.
+    upload.stop();
+    download.stop();
+    socket.close();
+    session.close();
+
+    if (listener != null) {
+      // We had a functional socket long enough to record statistics.
+      // Report the BYTES event : value = total transfer over the lifetime of a socket
```

Comment:
Fix formatting of VALUE field in comment.

Reply:
Done